Packs/AWS_SystemManager/Integrations/AWSSystemManager/AWSSystemManager.py

```diff
@@ -0,0 +1,1612 @@
+                "End Time",
+                "Automation Execution Status",
+                "Mode",
+                "Executed By",
+            ],
+        ),
+    )
+
+
+def list_automation_executions_command(
+    args: dict[str, Any],
+    ssm_client: "SSMClient",
+) -> list[CommandResults]:
+    """Lists AWS Systems Manager (SSM) automation executions.
+
+    Args:
+    ----
+        args (dict): A dictionary containing command arguments.
+            - limit (int, optional): The maximum number of results to return (default is 50).
+            - next_token (str, optional): A token to continue listing executions from a previous query.
+
+        ssm_client (SSMClient): An instance of the AWS Systems Manager (SSM) client.
+
+    Returns:
+    -------
+        list[CommandResults]: A list of objects containing the results of the command.
+        if next_token provide in the response, the first CommandResults in the list will contain the next token.
+    """
+    kwargs: "DescribeAutomationExecutionsRequestRequestTypeDef" = {
+        "MaxResults": arg_to_number(args.get("limit", 50)) or 50,
+    }
+    kwargs = update_if_value(args, kwargs, {"next_token": "NextToken"})
+
+    response = ssm_client.describe_automation_executions(**kwargs)
+    response = convert_datetime_to_iso(response)
+
+    command_results: list[CommandResults] = []
+    if next_token := response.get("NextToken"):
+        command_results.append(
+            next_token_command_result(next_token, "AutomationExecutionNextToken"),
+        )
+    automation_execution_list = response["AutomationExecutionMetadataList"]
+    command_results.append(
+        CommandResults(
+            outputs=automation_execution_list,
+            outputs_key_field="AutomationExecutionId",
+            outputs_prefix="AWS.SSM.AutomationExecution",
+            readable_output=tableToMarkdown(
+                name="AWS SSM Automation Executions",
+                t=[
+                    parse_automation_execution(automation)
+                    for automation in automation_execution_list
+                ],
+                headers=[
+                    "Automation Execution Id",
+                    "Document Name",
+                    "Document Version",
+                    "Start Time",
+                    "End Time",
+                    "Automation Execution Status",
+                    "Mode",
+                    "Executed By",
+                ],
+            ),
+        ),
+    )
+    return command_results
+
+
+def validate_target_arguments(args: dict[str, Any]) -> None:
+    if args.get("target_parameter_name") and not args.get("target_key"):
+        raise DemistoException(
+            "You must provide a target_key when specifying a target_parameter_name."
+        )
+    if args.get("target_key") and not args.get("target_parameter_name"):
+        raise DemistoException(
+            "You must provide a target_parameter_name when specifying a target_key."
+        )
```

Comment:
this code can be cut by half. 
1. use `sum(bool(x), bool(y)) == 1` instead of `x and not y or y and not x`
2. instead of repeated `if args.get("target_values")`, assign it to a variable and reuse it. (same for others)